### ISSUE TITLE
Command Output for use in PR Comment

### ISSUE THREAD

**Issue by lauragalera** [astral-sh/ruff-action#28]
I’m trying to use the ruff-action in my GitHub Actions workflow, which uses `format` and `check --fix`, to validate our python files.  I want to capture the output of these commands and include them in a comment on the pull request, but the only output set is the version.

My idea would be to do something like

```
      - name: 📬 Add output to PR
        run: |
          echo "#### Ruff Changes" > ruff_output.txt
          echo "<details><summary>Check</summary>" >> ruff_output.txt
          echo "" >> ruff_output.txt
          echo '```' >> ruff_output.txt
          echo "${{ steps.check.outputs.command-output }}" >> ruff_output.txt
          echo '```' >> ruff_output.txt
          echo "" >> ruff_output.txt
          echo "</details>" >> ruff_output.txt
          gh issue comment ${{ github.event.number }} --body-file ruff_output.txt
        env:
          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

However, I've noticed that the only output the action exposes is the version of ruff. I need to capture the actual formatting and linting results, but I'm unsure how to do this.

**Comment by eifinger** [comment]
Have you tried

```yaml
- uses: astral-sh/ruff-action@v2
  with:
    args: "format --check --output-file ruff_output.txt --output-format full"
```

I also added problem matchers in https://github.com/astral-sh/ruff-action/pull/26 which I will release in a new version once a few other things are done as well. This annotates the errors directly in the PR:

 ![Image](https://github.com/user-attachments/assets/5362f76c-3394-41e1-a80a-c4dfd130221b)

**Comment by lauragalera** [comment]
I tried your suggestion, and I get an error: `unexpected argument '--output-file' found`, same for `--output-format`.

We apply the format directly, but for the check, we would be interested in having a report of fails in a comment or even better, having annotations in the same code. 

I hope the release comes soon! Thank you

**Comment by lauragalera** [comment]
Indeed, it worked! This will do until the feature of error annotations is out. Thanks ^^

**Comment by eifinger** [comment]
@lauragalera have you tried with the new release?

**Comment by lauragalera** [comment]
Sorry, I was absent until today.
I'd like to test it out! Could you provide an example of how to use the matchers in the new version? I want to use it for the command `check`. Thank you!

**Comment by lauragalera** [comment]
I can see the annotations in the workflow run, but when I navigate back to the PR, they don't appear in the respective code. Am I missing something?
<img width="867" alt="Image" src="https://github.com/user-attachments/assets/9c3c9d9c-604f-446c-98ca-8a4b64d81c96" />.

**Comment by eifinger** [comment]
Is this somthing public you can point me to? Otherwise, it's in the "Files changed" tab. e.g. here for a PR in this repo:
https://github.com/astral-sh/ruff-action/pull/39/files

<img width="974" alt="Image" src="https://github.com/user-attachments/assets/757bc48b-1c50-4cf5-acbc-d9624e7893e2" />

**Comment by lauragalera** [comment]
Yes, that repo is private, but I created [this public one](https://github.com/lauragalera/demo-ruff-action/pull/4) for testing purposes and the annotations appear in the code! It must be a problem on my end. I confirm the feature works!

**Comment by eifinger** [comment]
Mhh maybe you need to set a permission https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token

**Comment by eifinger** [comment]
> [@eifinger](https://github.com/eifinger) How do you get the format error as annotations? `ruff format` does not have `--output-format`.

The above annotations are the result of
https://github.com/astral-sh/ruff-action/blob/9bdf1198e71d5d61dd732610fb2b1e80798a13c3/.github/workflows/test.yml#L223

**Comment by mschoettle** [comment]
Thanks. It's weird. When I run `ruff format --check` with an environment variable `RUFF_OUTPUT_FORMAT=github` like is done by the action I do not get the same output (basically missing the "Error:" annotation in the output):

```shell
Run uv run ruff format --check
  uv run ruff format --check
  shell: /usr/bin/bash -e {0}
  env:
    RUFF_OUTPUT_FORMAT: github
    UV_CACHE_DIR: /home/runner/work/_temp/setup-uv-cache
Would reformat: db_management/utility.py
1 file would be reformatted, 67 files already formatted
Error: Process completed with exit code 1.
```

**Comment by mschoettle** [comment]
Found out that it works here due to #12

**Comment by lauragalera** [comment]
@eifinger,  have you changed something regarding code annotations? I just saw it is not working and I haven't changed the workflow. As you can see here https://github.com/lauragalera/demo-ruff-action/pull/7 no annotations were added but the action output errors https://github.com/lauragalera/demo-ruff-action/actions/runs/15633154164. Using latest version v3.4.0
